plugins/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/condition/ApplicationPodIsRunning.java

```diff
@@ -55,6 +55,9 @@ public boolean test() {
 			return false;
 		}
 		
+		// TODO: this FLAWED: it assumes that all pods that it finds within a
+		// project are the pods for the application that a test wants to wait
+		// for
```

Comment:
The wait condition is meant for a connection with default project and a single application (service). For extension of this wait condition to be usable with a project which have more pods, the constructor would have to have parameters. I would rather update javadoc of the wait condition class to be more specific about usage.